core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -330,19 +332,21 @@ object SparkSubmit extends CommandLineUtils {
     args.archives = Option(args.archives).map(resolveGlobPaths(_, hadoopConf)).orNull
 
     // In client mode, download remote files.
+    var localPrimaryResource: String = null
+    var localJars: String = null
+    var localPyFiles: String = null
```

Comment:
ok, I guess yarn is special in that case.  Actually I see the help menu for --files and --archives is wrong for yarn as it says on the executors when it really goes to both executors and application master/driver.   The original feature was for mesos so which makes  more sense there. 

And really I guess most people aren't even going to know about this feature so unless they stumble into it running yarn client mode most will just continue to specify things as local filesystem.  

@jerryshao you are right, like I mentioned above to take advantage of files/archives on yarn you would have to be in ./ which isn't trivial.  Unless someone requests it lets just drop it.

Sorry for the tangent, I'm fine with it as is.